src/lib.rs

```diff
@@ -134,6 +283,43 @@ pub trait FutureExt: Future + Sized {
 
 impl<F> FutureExt for F where F: Future + Sized {}
 
+impl<T> FfiPoll<T> {
+    /// Converts a [`std::task::Poll`] to the [`FfiPoll`]
+    ///
+    /// [`std::task::Poll`]: std::task::Poll
+    pub fn from_poll(poll: Poll<T>) -> Self {
+        match poll {
+            Poll::Ready(r) => Self::Ready(r),
+            Poll::Pending => Self::Pending,
+        }
+    }
+    /// Converts a [`FfiPoll`] back to the [`std::task::Poll`]
+    ///
+    /// [`std::task::Poll`]: std::task::Poll
+    pub fn into_poll(self) -> Poll<T> {
+        match self {
+            Self::Ready(r) => Poll::Ready(r),
+            Self::Pending => Poll::Pending,
+            Self::Panicked => {
+                // FFI panicked, so let us panic too.
+                panic!("FFI future unwinded.");
```

Comment:
This method is `into_*` which I think isn't expected to panic. (So did `From::from`, see below)
I'd suggest
```rust
pub struct PollPanicked { _private: () }
pub fn try_into_poll(self) -> Result<Poll<T>, PollPanicked> { ... }

impl TryFrom<FfiPoll<T>> for Poll<T> {
    type Err = PollPanicked;
    // fn try_from ...
}
```

And an explicit panic in `LocalBorrowingFfiFuture::poll` rather than in `into_poll`.
The semantic is *panic in polling is propagated* not *converting a result indicating a panic panics*.

BTW: Panic message could be `FFI future panicked` or `FFI future panicked during polling`, no trailing period. (Not sure which of `during`/`in`/`when` is correct here. I'm not an English native-speaker).

Reply:
Would you mind adding a `thiserror` dependency for generating `impl`s for the `PollPanicked` struct?

Reply:
Doesn't it just need few lines? Why would you need it?
```rust
#[derive(Debug)]
pub struct PollPanicked { _private: () }
impl std::fmt::Display for PollPanicked {
    fn fmt(&self, f: &mut fmt::Formatter<'_>) -> fmt::Result {
        f.write_str("PollPanicked")
    }
}
impl std::error::Error for PollPanicked {}
```